### PR TITLE
More signup fixes desktop 703

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers 

More fixes for signup flow.

* Remove back from the success screen
* Add waiting states to signup buttons
* fix waiting styles for buttons (not signup specific)

### Waiting styles
Before
screen shot 2016-03-23 at 5 28 27 pm
After
screen shot 2016-03-23 at 6 39 06 pm

### A couple other potential bugs

We should discuss if we should invest time fixing these now

* Clear error fields when the user starts typing in the input boxes
  * We could fix this at the component level or parent level.

* Possible to create an account without seeing the paperkey
  * If there are certain errors in signup (like network timeout) it will retry later, but the gui doesn't get updated (it stays on the error screen) so the account is created in the service side, but the gui isn't notified about this. This will lead to outstanding rpcs too.

### CODE REVIEW

**Comment by chrisnojima** [comment]
We had a blocked ticket to not do the signup wait states as we need a
better solution than just the buttons. It's weird the spinners are so
broken for you they don't look like that for me

**Comment by chrisnojima** [comment]
Maybe it's a retina issue

**Comment by MarcoPolo** [comment]
@chrisnojima  You don't notice the broken spinner unless you add styles to the button. I changed the stylesheet to better show off the broken state, you can use those changes to see the broken state.

This is because the styles apply the the `FlatButton` component inside `Button`, but the spinner calculates it's height based on the height of `Button`. So if `FlatButton` has a margin top of 20 px, the calculated height of the spinner is 20px more than the visible `FlatButton` so it is rendered bigger and outside the bounds.

**Comment by MarcoPolo** [comment]
Is there a reason we shouldn't use the waiting states on buttons until we have a better solution? It seems a lot better to have a spinner button than nothing at all. Especially on spotty internet

**Comment by chrisnojima** [comment]
It's fine but might just be removed

**Comment by gabriel** [comment]
We could make waiting state not have a spinner too (so it's just a disabled state)... sometimes having too many spinners can be visually overwhelming...

+1 for Clear error fields when the user starts typing in the input boxes

**Comment by chrisnojima** [comment]
I believe there's a ticket for clearing errors already

**Comment by cjb** [comment]
It's DESKTOP-787.

File: shared/login/signup/success/index.render.desktop.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ export default class Render extends Component {
 
   render () {
     return (
-      <Container onBack={this.props.onBack} style={styles.container}>
```
Comment (chrisnojima) [inline shared/login/signup/success/index.render.desktop.js]: we could just keep container (and the styling) so its more consistent and if onBack is null it just doens't render the back button

**Comment by chrisnojima** [comment]
:+1: %c

**Pull request merged**